### PR TITLE
211203 wifi evt tc

### PR DESCRIPTION
**Pull request by pillip8282**

### CODE REVIEW

File: apps/examples/wifi_manager/wm_test/wm_test_event_tc.c [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+static void wm_cb_sta_connected(wifi_manager_cb_msg_s msg, void *arg);
+static void wm_cb_sta_disconnected(wifi_manager_cb_msg_s msg, void *arg);
+static void wm_cb_softap_sta_join(wifi_manager_cb_msg_s msg, void *arg);
+static void wm_cb_softap_sta_leave(wifi_manager_cb_msg_s msg, void *arg);
+static void wm_cb_scan_done(wifi_manager_cb_msg_s msg, void *arg);
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	wm_cb_sta_connected,
+	wm_cb_sta_disconnected,
+	wm_cb_softap_sta_join,
```
Comment (davidfather) [inline apps/examples/wifi_manager/wm_test/wm_test_event_tc.c:50]: wm_cb_softap_sta_joined and wm_cb_softap_sta_left might be better for consistency in naming.

File: os/include/tinyara/net/if/wifi.h [hunk 2]
```diff
@@ -27,6 +27,9 @@
 #define TRWIFI_PASSPHRASE_LEN		  64
 #define TRWIFI_CBK_MSG_INITIALIZER {0, {0,}, NULL}
 
+/*  reason */
+#define TRWIFI_REASON_UNKNOWN 0
```
Comment (davidfather) [inline os/include/tinyara/net/if/wifi.h:31]: @pillip8282 I can't find where this definition is used.

File: os/drivers/wireless/virtual/vdev_handler.c [hunk 3]
```diff
@@ -28,7 +28,7 @@
 #include <time.h>
 #include <tinyara/lwnl/lwnl.h>
 #include <tinyara/net/if/wifi.h>
-#include <tinyara/net/if/wifi_types.h>
+//#include <tinyara/net/if/wifi_types.h>
```
Comment (davidfather) [inline os/drivers/wireless/virtual/vdev_handler.c:31]: Why don't you remove this line?

File: apps/examples/wifi_manager/wm_test/wm_test_event_tc.c (hunk 1 above)
Comment (pillip8282) [reply apps/examples/wifi_manager/wm_test/wm_test_event_tc.c:50]: Thanks I'll modify as your recommendation.

File: os/include/tinyara/net/if/wifi.h (hunk 2 above)
Comment (pillip8282) [reply os/include/tinyara/net/if/wifi.h:31]: vdev_handler.c uses this value to test correctness of event information from wi-fi driver.
More definition will be added later.

File: os/drivers/wireless/virtual/vdev_handler.c (hunk 3 above)
Comment (pillip8282) [reply os/drivers/wireless/virtual/vdev_handler.c:31]: I forgot to remove the line. I'll delete the line

**Review by an4967 — APPROVED** [review]
LGTM

**Pull request merged**